### PR TITLE
Feature 607 provide pds zap integration

### PR DESCRIPTION
**Pull request by Jeeppler**
OWASP ZAP + PDS:

- Helm chart
- Docker container
- Documentation

closes: #607

### CODE REVIEW

**Review by winzj — CHANGES_REQUESTED** [review]
Please see my comments.

**Review by winzj — APPROVED** [review]
Looks good to me.

File: sechub-pds-solutions/owasp-zap/docker-compose_pds_owasp_zap_cluster.yaml [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+            - .env-cluster
+        networks:
+            - "internal"
+        ports:
+            - "127.0.0.1:5432:5432"
+        
+networks:
+    internal:
+    sechub:
+      name: "sechub"
```
Comment (sven-dmlr) [inline sechub-pds-solutions/owasp-zap/docker-compose_pds_owasp_zap_cluster.yaml:46]: pls fix identation:
```suggestion
        name: "sechub"
```

File: sechub-pds-solutions/owasp-zap/helm/pds-owasp-zap/templates/deployment.yaml [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+                    - name: S3_ACCESSKEY
+                      value: {{ .Values.storage.s3.accesskey }}
+                    - name: S3_SECRETKEY
+                      value: {{ .Values.storage.s3.secretkey }}
+                    {{- else if .Values.storage.sharedVolume.enabled }}
+                    - name: SHARED_VOLUME_UPLOAD_DIR
+                      value: "{{ .Values.storage.sharedVolume.upload.dir }}"
+                    {{- end}}
+                  ports:
+                    - containerPort: 8444
```
Comment (sven-dmlr) [inline sechub-pds-solutions/owasp-zap/helm/pds-owasp-zap/templates/deployment.yaml:86]: Please add startup+liveness probes:
```suggestion
                    - name: pds-https-port
                      containerPort: 8444
                  startupProbe:
                    httpGet:
                      scheme: HTTPS
                      path: /api/anonymous/check/alive
                      port: pds-https-port
                    periodSeconds: 1
                    failureThreshold: 300
                    # probe every 1s x 300 = 5 mins before restart of container
                    successThreshold: 1
                    timeoutSeconds: 1
                  livenessProbe:
                    httpGet:
                      scheme: HTTPS
                      path: /api/anonymous/check/alive
                      port: pds-https-port
                    periodSeconds: 5
                    failureThreshold: 3
                    successThreshold: 1
                    timeoutSeconds: 3
```

**Review by sven-dmlr — CHANGES_REQUESTED** [review]
Please see comments.

Looking forward to see PDS with OWASP-Zap running. :-)

File: sechub-pds-solutions/owasp-zap/10-create-ubuntu-image.sh [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+  exit 1
+fi
+
+if [[ -z "$BASE_IMAGE" ]]
+then
+    BASE_IMAGE="ubuntu:20.04"
+fi
+
+echo ">> Base image: $BASE_IMAGE"
+docker build --pull --no-cache --build-arg BASE_IMAGE=$BASE_IMAGE --tag "$REGISTRY:$VERSION" --file docker/Owasp-Zap-Ubuntu.dockerfile docker/
```
[the remark this answers is not in the document]
Comment (Jeeppler) [reply sechub-pds-solutions/owasp-zap/10-create-ubuntu-image.sh:26]: A tag is just a label which can move, just like the special label `latest`.

For example, Ubuntu 20.04 was released in April of 2020. However, at the time of writing, the latest Ubuntu 20.04 image is tagged with: [20.04, focal-20220404, focal, latest](https://hub.docker.com/_/ubuntu/). The Ubuntu `focal-20220404` tag shows the image was build on the 4th of April 2022. Meaning, the tag `20.04` points to an image which was build two years after the official release.

As a result, I don't think it is a good idea to change the behavior of the script based on the tags.

Furthermore, the problem which you are referring too is specific to organizations which do not have a DockerHub subscription or surpassed their limits.

> Then this error may occur: "toomanyrequests: You have reached your pull rate limit."

There are a number of solutions on how to fix the `toomanyrequests` issue, without changing the behavior of this script.

**Review by sven-dmlr — APPROVED** [review]

**Pull request merged**